src/main/java/com/chopshop166/chopshoplib/RobotUtils.java

```diff
@@ -128,4 +134,94 @@ public static double sps(final double value) {
     public static BooleanSupplier not(final BooleanSupplier func) {
         return () -> !func.getAsBoolean();
     }
+
+    /**
+     * Get the MAC address
+     * 
+     * @return The robot's MAC address as a string.
+     */
+    public static String getMACAddress() {
+        try {
+            final NetworkInterface iface = NetworkInterface.getByName("eth0");
+            final byte[] mac = iface.getHardwareAddress();
+
+            if (mac == null) { // happens on windows sometimes
+                throw new SocketException();
+            }
+
+            final StringJoiner sb = new StringJoiner("-");
+            for (final byte octet : mac) {
+                sb.add(String.format("%02X", octet));
+            }
+            return sb.toString();
+        } catch (SocketException e) {
+            return "SocketException";
+        }
+    }
+
+    /**
+     * Get a RobotMap for the given name.
+     * 
+     * @param rootClass The root class object that the map derives from.
+     * @param pkg       The package to look in.
+     */
+    public static <T> T getRobotMap(final Class<T> rootClass, final String pkg) {
```

Comment:
Can we make this take `T` and get `T.class` inside this function? I would like to hide as much of the boilerplate we can.

Reply:
I wish, but the way Java generics work says no - once you get inside the method you can't get a reference to the class object for a generic parameter

Reply:
That's unfortunate. Luckily the example you provide is straightforward.